### PR TITLE
[sdl2] Update to 2.30.7

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version.~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by erikyuzwa** [comment]
Thanks @oleg-derevenetz! 🔥

File: ports/sdl2/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "sdl2",
-  "version": "2.30.6",
+  "version": "2.30.7",
   "port-version": 2,
```
Comment (WangWeiLin-MV) [inline ports/sdl2/vcpkg.json:4]:
```suggestion
```

**Review by WangWeiLin-MV — DISMISSED** [review]

**Review by WangWeiLin-MV — APPROVED** [review]
The port installation tests pass with the following triplets:

* x64-linux

**Comment by BillyONeal** [comment]
Thanks for the update and removing the patch!

**Pull request merged**